test/TableTest.hpp

```diff
@@ -169,9 +183,75 @@ TEST_CASE("Insert and select bigint.")
     REQUIRE(select_query->RowCount() == 1);
     const auto& rows = select_query->Rows();
     const auto& values = rows[0].Columns();
-    REQUIRE(values[0].AsInt64() == -9223372036854775807);
-    REQUIRE(values[1].AsUInt64() == 9223372036854775807);
+
+    REQUIRE(values[0].AsInt64().has_value());
+    REQUIRE(values[0].AsInt64().value() == -9223372036854775807);
+    REQUIRE(values[1].AsInt64().has_value());
+    REQUIRE(values[1].AsUInt64().value() == 9223372036854775807);
 
     REQUIRE(values[0].AsStringView() == "-9223372036854775807");
     REQUIRE(values[1].AsStringView() == "9223372036854775807");
 }
+
+TEST_CASE("Test Null Insert")
+{
+    using namespace std::chrono_literals;
+    wing::ConnectionInfo connection { MYSQL_HOSTNAME, MYSQL_PORT, MYSQL_USERNAME, MYSQL_PASSWORD };
+    wing::Executor executor { std::move(connection) };
+
+    wing::Statement insert_stm {};
+    insert_stm << "INSERT INTO " << MYSQL_DATABASE << ".integers (b, ub) VALUES (NULL, 1515151515)";
+    auto insert_query = executor.StartQuery(std::move(insert_stm), 10s).value().get();
+    query_print_error(insert_query);
+    REQUIRE(insert_query->QueryStatus() == wing::QueryStatus::SUCCESS);
+    REQUIRE(insert_query->RowCount() == 1);
+
+    wing::Statement select_stm {};
+    select_stm << "SELECT b, ub FROM " << MYSQL_DATABASE << ".integers WHERE id = " << insert_query->LastInsertId();
+    auto select_query = executor.StartQuery(std::move(select_stm), 10s).value().get();
+    query_print_error(select_query);
+    REQUIRE(select_query->QueryStatus() == wing::QueryStatus::SUCCESS);
+    REQUIRE(select_query->RowCount() == 1);
+    const auto& rows = select_query->Rows();
+    const auto& values = rows[0].Columns();
+
+    REQUIRE_FALSE(values[0].AsInt64().has_value());
+    REQUIRE(values[1].AsInt64().has_value());
+}
+
+TEST_CASE("Test Throw Returns un-engaged optional")
```

Comment:
I don't think this test actually throws since you are using has_value() now?  maybe just a more appropriate test name